mlflow/tracking/client.py

```diff
@@ -333,6 +333,13 @@ def set_terminated(self, run_id, status=None, end_time=None):
         :param end_time: If not provided, defaults to the current time."""
         self._tracking_client.set_terminated(run_id, status, end_time)
 
+    def set_running(self, run_id, status=None):
```

Comment:
This API is a little odd, because we can pass a `status` other than `RUNNING` to an API called `set_running`. Could we instead expose a method called `update_run` that exposes the same parameters as the [underlying REST API](https://mlflow.org/docs/latest/rest-api.html#update-run)? Pretty similar to `set_terminated`, except that we don't pass an `end_time` if the user doesn't specify one

Reply:
Also note that this would be a new user-facing API, so we should mark this PR with rn/feature and add a release note

Reply:
Updated with sequence of changes discussed offline.